iot_packet_verifier/src/verifier.rs

```diff
@@ -0,0 +1,496 @@
+                self.config_server
+                    .enable_org(report.oui)
+                    .await
+                    .map_err(VerificationError::ConfigError)?;
+            } else {
+                invalid_packets
+                    .write(InvalidPacket {
+                        payload_size: report.payload_size,
+                        gateway: report.gateway.into(),
+                        payload_hash: report.payload_hash,
+                        reason: InvalidPacketReason::InsufficientBalance as i32,
+                    })
+                    .await
+                    .map_err(VerificationError::InvalidPacketWriterError)?;
+                self.config_server
+                    .disable_org(report.oui)
+                    .await
+                    .map_err(VerificationError::ConfigError)?;
+            }
+        }
+
+        Ok(())
+    }
+}
+
+pub fn payload_size_to_dc(payload_size: u64) -> u64 {
+    let payload_size = payload_size.max(24);
+    // perform a div_ciel function:
+    (payload_size + 24 - 1) / 24
+}
+
+#[async_trait]
+pub trait Debiter {
+    type Error;
+
+    async fn debit_if_sufficient(
+        &self,
+        payer: &PublicKeyBinary,
+        amount: u64,
+    ) -> Result<bool, Self::Error>;
+}
+
+#[async_trait]
+pub trait ConfigServer {
+    type Error;
+
+    async fn fetch_org(
+        &mut self,
+        oui: u64,
+        cache: &mut HashMap<u64, PublicKeyBinary>,
+    ) -> Result<PublicKeyBinary, Self::Error>;
+
+    async fn enable_org(&mut self, oui: u64) -> Result<(), Self::Error>;
+
+    async fn disable_org(&mut self, oui: u64) -> Result<(), Self::Error>;
+}
+
+// TODO: Move this somewhere else
+
+// Probably should change name to something like OrgClientCache to be more
+// consistent with BalanceCache
+pub struct CachedOrgClient {
+    pub keypair: Keypair,
+    pub enabled_clients: HashMap<u64, bool>,
+    pub client: OrgClient<Channel>,
+}
+
+impl CachedOrgClient {
+    pub fn new(client: OrgClient<Channel>, keypair: Keypair) -> Self {
+        CachedOrgClient {
+            keypair,
+            enabled_clients: HashMap::new(),
+            client,
+        }
+    }
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum OrgClientError {
+    #[error("Rpc error: {0}")]
+    RpcError(#[from] tonic::Status),
+    #[error("Crypto error: {0}")]
+    CryptoError(#[from] helium_crypto::Error),
+}
+
+#[async_trait]
+impl ConfigServer for CachedOrgClient {
+    type Error = OrgClientError;
+
+    async fn fetch_org(
+        &mut self,
+        oui: u64,
+        cache: &mut HashMap<u64, PublicKeyBinary>,
+    ) -> Result<PublicKeyBinary, Self::Error> {
+        if let Entry::Vacant(e) = cache.entry(oui) {
+            let req = OrgGetReqV1 { oui };
+            let pubkey =
+                PublicKeyBinary::from(self.client.get(req).await?.into_inner().org.unwrap().owner);
+            e.insert(pubkey);
+        }
+        Ok(cache.get(&oui).unwrap().clone())
```

Comment:
stray unwraps?

Reply:
This is guaranteed to never panic, an insert always occurs before this if no entry was found